ord_schema/scripts/migrate_schema.py

```diff
@@ -0,0 +1,222 @@
+
+
+def migrate_stirring_conditions(old: reaction_old_pb2.StirringConditions,
+                                new: reaction_pb2.StirringConditions):
+    """Migrates a StirringConditions message."""
+    new.type = old.method.type
+    new.details = old.method.details
+    migrate_messages(old, new, ('rate',))
+
+
+def migrate_illumination_conditions(
+        old: reaction_old_pb2.IlluminationConditions,
+        new: reaction_pb2.IlluminationConditions):
+    """Migrates an IlluminationConditions messsage."""
+    new.type = old.type.type
+    new.details = old.type.details
+    migrate_messages(old, new, ('peak_wavelength', 'distance_to_vessel'))
+    migrate_scalars(old, new, ('color',))
+
+
+def migrate_electrochemistry_conditions(
+        old: reaction_old_pb2.ElectrochemistryConditions,
+        new: reaction_pb2.ElectrochemistryConditions):
+    """Migrates an ElectrochemistryConditions messsage."""
+    new.type = old.electrochemistry_type.type
+    new.details = old.electrochemistry_type.details
+    migrate_scalars(old, new, ('anode_material', 'cathode_material'))
+    migrate_messages(old, new,
+                     ('current', 'voltage', 'electrode_separation', 'cell'))
+    migrate_repeated_messages(old, new, ('measurements',))
+
+
+def migrate_flow_conditions(old: reaction_old_pb2.FlowConditions,
+                            new: reaction_pb2.FlowConditions):
+    """Migrates a FlowConditions messsage."""
+    new.type = old.flow_type.type
+    new.details = old.flow_type.details
+    migrate_scalars(old, new, ('pump_type',))
+    migrate_messages(old, new, ('tubing',))
+
+
+def migrate_setup(old: reaction_old_pb2.ReactionSetup,
+                  new: reaction_pb2.ReactionSetup):
+    """Migrates a ReactionSetup messsage."""
+    if old.HasField('vessel'):
+        migrate_vessel(old.vessel, new.vessel)
+    migrate_optional_scalars(old, new, ('is_automated',))
+    migrate_scalars(old, new, ('automation_platform',))
+    for key, value in old.automation_code.items():
+        migrate_message(value, new.automation_code[key])
+    migrate_messages(old, new, ('environment',))
+
+
+def migrate_conditions(old: reaction_old_pb2.ReactionConditions,
+                       new: reaction_pb2.ReactionConditions):
+    """Migrates a ReactionConditions messsage."""
+    migrate_messages(old, new, ('temperature', 'pressure'))
+    if old.HasField('stirring'):
+        migrate_stirring_conditions(old.stirring, new.stirring)
+    if old.HasField('illumination'):
+        migrate_illumination_conditions(old.illumination, new.illumination)
+    if old.HasField('electrochemistry'):
+        migrate_electrochemistry_conditions(old.electrochemistry,
+                                            new.electrochemistry)
+    if old.HasField('flow'):
+        migrate_flow_conditions(old.flow, new.flow)
+    migrate_optional_scalars(old, new, ('reflux', 'conditions_are_dynamic'))
+    if old.HasField('pH'):
+        new.ph = old.pH  # Lowercase now.
```

Comment:
Again, not necessarily opposed, just don't remember reasoning for lowercase.

Reply:
See above.